src/core/server/elasticsearch/version_check/ensure_es_version.ts

```diff
@@ -142,6 +142,21 @@ function compareNodes(prev: NodesVersionCompatibility, curr: NodesVersionCompati
   );
 }
 
+function obtainNodesInfo(client: ElasticsearchClient): Observable<NodesInfo> {
+  const infoPromise: Promise<NodesInfo> = client.nodes.info({
+    filter_path: ['nodes.*.version', 'nodes.*.http.publish_address', 'nodes.*.ip'],
+  });
+
+  return infoPromise
+    ? from(infoPromise).pipe(
```

Comment:
Why this check? `client.nodes.info` always return a value. Are we adapting production code to adapt for testing environment rather than the other way around here?

Reply:
If I recall correctly, at some point during an integration test, the `client.nodes.info()` was returning `undefined` instead of a Promise. I don't remember the exact scenario but that's the reason I changed this. Perhaps it is linked to the fact that we start and stop servers very quickly for some tests, not giving this request enough time to complete.

I'll rollback this change to expose the error again.